### PR TITLE
:memo: Added considerations and future work doc

### PR DESCRIPTION
**Pull request by tchataigner**
This PR introduces documentation for considerations while building the SDK and already identified future work that will need to be conducted.

### CODE REVIEW

File: content/en/docs/reference/considerations.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+> You can find the initial grant proposal for the high level Rust SDK 1.0 [here](https://github.com/filecoin-project/devgrants/issues/562)
+
+## External considerations
+
+The goal for this SDK was to create a set of tools that would help actors developer to get on building
+over the Filecoin network without having to think about overhead code used to comply to the network standard.
+
+In that sense the SDK is meant to be:
+- **Simple**: Entry-level Rust developer should be able to use the SDK to produce actors that can be
+ran in the FVM.
```
Comment (PhilippeMts) [inline content/en/docs/reference/considerations.md]:
```suggestion
run in the FVM.
```

File: content/en/docs/reference/future-work.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+---
+
+> If you have any ideas for iterations that you can not find in the list down below please feel free
+> to open an issue on the [repository](https://github.com/polyphene/fvm-rs-sdk).
+
+## _fvm_actor_
+
+- Developers should be able to develop their own invoke function without using `#[fvm_actor]` and still
+feel as less overhead as possible. Different design should be studied (macros for code  generation of
+serde/dispatch/etc , code injection at dedicated places in the generated invoke ...) and one implemented.
```
Comment (PhilippeMts) [inline content/en/docs/reference/future-work.md]:
```suggestion
serde/dispatch/etc., code injection at dedicated places in the generated invoke ...) and one implemented.
```

File: content/en/docs/reference/future-work.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+draft: false
+images: []
+menu:
+docs:
+parent: "reference"
+weight: 400
+toc: true
+---
+
+> If you have any ideas for iterations that you can not find in the list down below please feel free
```
Comment (PhilippeMts) [inline content/en/docs/reference/future-work.md]:
```suggestion
> If you have any idea for iterations that you can not find in the list down below please feel free
```

**Review by PhilippeMts — CHANGES_REQUESTED** [review]
Thanks for this update @tchataigner .

You'll find a few typos you may fix.

One last comment: the _Get Started_ button on the front page may link to rather link to _Considerations_ instead of _Usage_ which weight is higher in the ToC. 👍

**Comment by tchataigner** [comment]
> One last comment: the Get Started button on the front page may link to rather link to Considerations instead of Usage which weight is higher in the ToC. 👍

Yes you are right ! Changing that !

**Review by PhilippeMts — APPROVED** [review]

**Pull request closed without merging**